### PR TITLE
Merge xcode11 into d16-3.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Comment by dalexsoto** [comment]
LGTM so far 👍

File: Make.config [hunk 1]
```diff
@@ -11,7 +11,8 @@ $(TOP)/Make.config.inc: $(TOP)/Make.config
 	@if test -d $(TOP)/../maccore; then printf "ENABLE_XAMARIN=1\n" >> $@; echo "Detected the maccore repository, automatically enabled the Xamarin build"; fi
 	@# Build from source for Xcode 11 support until mono has packages
 	@# TODO: to be removed when we can use the mono archive again.
-	@printf "MONO_BUILD_FROM_SOURCE=1\n" >> $@; echo "Disabled the packaged mono build, since it's currently not supported in the xcode11 support branch."
+	#@printf "MONO_BUILD_FROM_SOURCE=1\n" >> $@; echo "Disabled the packaged mono build, since it's currently not supported in the xcode11 support branch."
+	#@ REVIEW: mono doesn't build from source in the 2019-06 branch (when we do it), so temporarly use packages again.
```
[the remark this answers is not in the document]
Comment (rolfbjarne) [reply Make.config]: I've opened this: https://github.com/mono/mono/issues/16821, and I also tried a very ugly workaround to see if we can build it ourselves using the xcode11 toolchain.

File: Make.config [hunk 2]
```diff
@@ -9,6 +9,10 @@ $(TOP)/Make.config.inc: $(TOP)/Make.config
 	@printf "MAC_COMMIT_DISTANCE:=$(shell LANG=C; export LANG && git --git-dir $(TOP)/.git log `git --git-dir $(TOP)/.git blame -- ./Make.versions HEAD | grep MAC_PACKAGE_VERSION= | sed 's/ .*//' `..HEAD --oneline | wc -l | sed 's/ //g')\n" >> $@
 	@if which ccache > /dev/null 2>&1; then printf "ENABLE_CCACHE=1\nexport CCACHE_BASEDIR=$(abspath $(TOP)/..)\n" >> $@; echo "Found ccache on the system, enabling it"; fi
 	@if test -d $(TOP)/../maccore; then printf "ENABLE_XAMARIN=1\n" >> $@; echo "Detected the maccore repository, automatically enabled the Xamarin build"; fi
+	@# Build from source for Xcode 11 support until mono has packages
+	@# TODO: to be removed when we can use the mono archive again.
+	#@printf "MONO_BUILD_FROM_SOURCE=1\n" >> $@; echo "Disabled the packaged mono build, since it's currently not supported in the xcode11 support branch."
+	#@ REVIEW: mono doesn't build from source in the 2019-06 branch (when we do it), so temporarly use packages again.
```
[the remark this answers is not in the document]
Comment (rolfbjarne) [reply Make.config]: Yes, eventually we will, but probably not in this PR.

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
CI redness is not bad enough to prevent the merge:

* xharness tries to run tests on device in public Jenkins, and this shows up as failed tests. These tests should not be run in the first place, so these failures are not real failures. All the tests that should be executed were, and those all passed.
* The breaking changes in the API diff are expected.
* The test failure on macOS 10.15 is known and will be fixed the next time xcode11 is merged into d16-3.
* There's one test failure due to https://github.com/xamarin/maccore/issues/581.
* The device tests failed to run because of a bug in xharness; I ran a few smoke device tests locally and things seemed to work fine.

**Pull request merged**